### PR TITLE
Take into account if a glacier object is restored for s3 downloads

### PR DESCRIPTION
**Pull request by jamesls**
If a glacier object has been restored, we should be able
to download the file.

Fixes a regression introduced when we started ignoring the
GLACIER storage class.

This doesn't fix the issue for a recursive downloads/sync,
but that isn't possible because the Restore data isn't
in a ListObjects request.

cc @kyleknap @mtdowling @rayluo @JordonPhillips

### CODE REVIEW

File: awscli/customizations/s3/fileinfo.py [hunk 1]
```diff
@@ -279,10 +280,24 @@ def is_glacier_compatible(self):
...
-            if response_data.get('StorageClass') == 'GLACIER':
+            if response_data.get('StorageClass') == 'GLACIER' and \
+                    not self._is_restored(response_data):
                 return True
         return False
 
+    def _is_restored(self, response_data):
+        # Returns True is this is a glacier object that has been
+        # restored back to S3.
+        # 'Restore' looks like: 'ongoing-request="false", expiry-date="..."'
```
Comment (rayluo) [inline awscli/customizations/s3/fileinfo.py]: So is the following effect equivalent to `return r'ongoing-request="false"' in response_data.get('Restore', '')`?

File: awscli/customizations/s3/fileinfo.py [hunk 2]
```diff
@@ -279,10 +280,24 @@ def is_glacier_compatible(self):
...
+        result = re.search(r'ongoing-request="(.+?)"',
+                           response_data.get('Restore', ''))
+        if result is None:
+            return False
+        ongoing_request_value = result.group(1)
+        if ongoing_request_value == 'false':
+            return True
+        return False
```
Comment (rayluo) [inline awscli/customizations/s3/fileinfo.py]: So is the above effect equivalent to this line?

    return r'ongoing-request="false"' in response_data.get('Restore', '')

Comment (jamesls) [inline awscli/customizations/s3/fileinfo.py]: Nice.  Good catch, I'll update this.

**Comment by jamesls** [comment]
@rayluo Feedback incorporated.

**Comment by rayluo** [comment]
Thanks! :ship:

File: awscli/customizations/s3/fileinfo.py [hunk 3]
```diff
@@ -1,4 +1,5 @@
 import os
+import re
```
Comment (kyleknap) [inline awscli/customizations/s3/fileinfo.py]: Unused import

**Comment by kyleknap** [comment]
Looks good. You now have an unused import ``re``. Otherwise :ship:

**Comment by jamesls** [comment]
Rebased, squashed, merged via https://github.com/aws/aws-cli/commit/21b6bd228202343018fba2e9a554c917751895e1

**Pull request closed without merging**